pkg/sql/tablemetadatacache/table_metadata_updater.go

```diff
@@ -40,38 +45,83 @@ type tableMetadataDetails struct {
 	StatsLastUpdated *time.Time `json:"stats_last_updated"`
 }
 
-// newTableMetadataUpdater creates a new tableMetadataUpdater.
-func newTableMetadataUpdater(ie isql.Executor) *tableMetadataUpdater {
-	return &tableMetadataUpdater{ie: ie, upsertQuery: newTableMetadataBatchUpsertQuery(tableBatchSize)}
+var _ tablemetadatacacheutil.ITableMetadataUpdater = &tableMetadataUpdater{}
+
+// newTableMetadataUpdater creates a new iTableMetadataUpdater.
+var newTableMetadataUpdater = func(
+	job *jobs.Job,
+	metrics *TableMetadataUpdateJobMetrics,
+	ie isql.Executor,
+	testKnobs *tablemetadatacacheutil.TestingKnobs) *tableMetadataUpdater {
+	return &tableMetadataUpdater{
+		ie:      ie,
+		metrics: metrics,
+		updateProgress: func(ctx context.Context, progress float32) {
+			updateProgress(ctx, job, progress)
+		},
+		testKnobs: testKnobs,
+	}
+}
+
+// RunUpdater prunes then updates table_metadata and records metrics for the run.
```

Comment:
Instead we can add a comment here referencing that RunUpdater implements ITableMetadataUpdater.

Reply:
Done